helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/CodeTransformations.java

```diff
@@ -548,6 +553,38 @@ public static String removeAvroGeneratedAnnotation(String code, AvroVersion minS
     return AVROGENERATED_ANNOTATION_PATTERN.matcher(code).replaceAll("");
   }
 
+  /**
+   * MODEL$ was introduced as a static field in the generated class, starting 1.8, which indicates the conversions
+   * applicable for a specific record. However, starting 1.9, avro uses reflection to look up this field, which will throw
+   * a {@link ReflectiveOperationException exception for records generated from older version. This results in  performance
+   * degradation.
+   * This methods avoids the exception by introducing this field in older versions of the generated record.
+   *
+   * @param code avro generated code that may or may not have the MODEL$ declaration
+   * @param minSupportedVersion lowest avro version under which the generated code should work
+   * @param maxSupportedVersion highest avro version under which the generated code should work
+   * @return code where MODEL$ exists for avro versions expecting it at runtime (>= 1.9)
+   */
+  public static String pacifyModel$Delcaration(String code, AvroVersion minSupportedVersion, AvroVersion maxSupportedVersion) {
+    if (minSupportedVersion.laterThan(AvroVersion.AVRO_1_7)) { // min >= 1.8
+      return code;
+    }
+    if (maxSupportedVersion.earlierThan(AvroVersion.AVRO_1_8)) { // max < 1.8
+      // remove any model annotation as it will not be used
+      return MODEL_DECL_PATTERN.matcher(code).replaceAll("");
+    } else {
+      // If MODEL$ is already present in the code, replace it with SpecificData#get() instead of SpecificData ctor, which was
+      // made public only from avro 1.7
+      if (MODEL_DECL_PATTERN.matcher(code).find()) {
+        return MODEL_DECL_PATTERN.matcher(code).replaceAll(Matcher.quoteReplacement(MODEL_DECL_REPLACEMENT));
```

Comment:
i dont think we should do this "simplification" - unless pre-existing MODEL$s use API which will not be supported under old avro - and then we should do it based on the min supported avro version

Reply:
do you mean this condition should be:
`minSupportedVersion < avro_1.7 && MODEL_DECL_PATTERN found in code`?